### DISCUSSION TITLE
go 1.18 Question: Could sqlboiler move from code generation to generics?

### DISCUSSION THREAD

**Discussion by jarrettv** [aarondl/sqlboiler#1107] [Ideas]
I'm currently evaluating ORMs and I'm looking for a generic solution that minimizes string constants so if the data model changes I can easily identify all the places to change the code. Is it too early to ask?

**Comment by stephenafamo** [comment]
That is the plan for v5.

There will likely be a bit of code generation at least to generate the base struct models for the tables that match the columns.  
However, it would likely be based on a Generic type.
